core/src/test/scala/unit/kafka/server/AbstractFetcherThreadTest.scala

```diff
@@ -123,4 +124,77 @@ class AbstractFetcherThreadTest {
       new DummyFetchRequest(partitionMap.map { case (k, v) => (k, v.offset) }.toMap)
   }
 
+
+  @Test
+  def testFetchRequestCorruptedMessageException() {
+    val partition = new TopicPartition("topic", 0)
+    val fetcherThread = new TestFetcherThread("test", "client", new BrokerEndPoint(0, "localhost", 9092))
+
+    fetcherThread.start()
+
+    // add one partition for fetching
+    fetcherThread.addPartitions(Map(partition -> 0L))
+
+    // wait until fetcherThread finishes the work
+    TestUtils.waitUntilTrue(() => (fetcherThread.fetchCount > 3),
+      "Failed waiting for fetcherThread tp finish the work")
+
+    fetcherThread.shutdown()
+
+    assertTrue(fetcherThread.logEndOffset == 2);
+  }
+
+  class TestFetcherThread(name: String,
```

Comment:
Can we rename this to ```TestFetcherThreadForCorruptedMessage```? Also it might be useful for future developers to understand the purpose of this test by adding some comments about what we are trying to test here.